aws/gd.go

```diff
@@ -0,0 +1,74 @@
+package aws
+
+import (
+	"context"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/guardduty"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/go-commons/errors"
+)
+
+// getAllGuardDutyDetectors returns active GuardDuty detectors in a given region.
+func getAllGuardDutyDetectors(session *session.Session, excludeAfter time.Time) ([]*string, error) {
+	svc := guardduty.New(session)
+
+	ctx := context.Background()
+	var detectors []*string
+
+	err := svc.ListDetectorsPagesWithContext(ctx, &guardduty.ListDetectorsInput{},
```

Comment:
Is GuardDuty enabled in every AWS region? If not, we'd need to handle the error when it's not enabled, and skip that region.

Reply:
I might be misunderstanding what you mean by enabled, but the method does not return an error if there are no GuardDuty detectors in a region, only for service API and SDK errors.